mozilla/fx-private-relay
# MPP-3894: Downgrade Typescript to 5.4.x

We're getting [an error in CircleCI](https://app.circleci.com/pipelines/github/mozilla/fx-private-relay/20165/workflows/4529d73e-2be4-43a7-8687-ac1cd8302977/jobs/214850?invite=true#step-109-166_13), and @groovecoder [can reproduce locally](https://github.com/mozilla/fx-private-relay/pull/5026#issuecomment-2344077620):

```
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=4.7.4 <5.5.0

YOUR TYPESCRIPT VERSION: 5.5.4

Please only submit bug reports when using the officially supported version.
```

This PR:

1. Removes the `override` section from `frontend/packages.json`, since overrides only work in the root package.json (https://github.com/npm/cli/issues/4517).
2. Add an override for typescript 5.4.5, the latest in the 5.4.x series
3. Uninstall and re-install packages to get them all on typescript 5.4.5,
4. Fix the return type of `formatRangeToParts` to get typescript passing

How to test:

1. Run `npm install`
2. Run `npm list typescript`. All versions should be 5.4.5. If not, you may need to remove `node_modules` and `frontend/node_modules`, then start over at step 1
3. In the `frontend` folder, run `npm run build` to build the frontend. There should be no errors.
4. Still in the `frontend` folder, run `npm run lint` to lint the frontend. There should be no warnings.
